planner/core/planbuilder.go

```diff
@@ -544,22 +550,28 @@ func (b *PlanBuilder) popSelectOffset() {
 	b.selectOffset = b.selectOffset[:len(b.selectOffset)-1]
 }
 
-// NewPlanBuilder creates a new PlanBuilder.
-func NewPlanBuilder(sctx sessionctx.Context, is infoschema.InfoSchema, processor *hint.BlockHintProcessor) *PlanBuilder {
+// NewPlanBuilderWithSetVarCollect creates a new PlanBuilder with setVarCollectProcessor.
+func NewPlanBuilderWithSetVarCollect(sctx sessionctx.Context, is infoschema.InfoSchema, processor *hint.BlockHintProcessor, setVarCollectProcessor *SetVarCollectProcessor) *PlanBuilder {
```

Comment:
Why we need to add a wrapper here?

Reply:
Because I need `setVarCollectProcessor` field in `PlanBuilder`.
The orignal `NewPlanBuilder` is invoked in many other files, and passing `nil` is annoying.

Reply:
The wrapper has removed. 
Only use `NewPlanBuilder` to create is better. 
You are right.